core/src/test/java/com/linecorp/armeria/server/throttling/RateLimitingThrottlingStrategyTest.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.throttling;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+
+import com.linecorp.armeria.common.Request;
+
+public class RateLimitingThrottlingStrategyTest {
+    @Test
+    public void rateLimit() throws Exception {
+        RateLimitingThrottlingStrategy<Request> strategy = new RateLimitingThrottlingStrategy<>(1);
+        assertThat(strategy.accept(null, null).get()).isEqualTo(true);
+        assertThat(strategy.accept(null, null).get()).isEqualTo(false);
+        assertThat(strategy.accept(null, null).get()).isEqualTo(false);
+
+        TimeUnit.SECONDS.sleep(1);
```

Comment:
I don't think this changes the flakiness much, we should add a `@VisibleForTesting` constructor that can take a mocked ratelimiter, or at least use awaitility here instead of a fixed sleep

Reply:
added VisibleForTesting constructor and inject mocked RateLimiter.